packages/sanity/src/structure/panes/document/documentPanel/documentViews/useConditionalToast.ts

```diff
@@ -32,5 +32,15 @@ export function useConditionalToast(params: ToastParams & {id: string; enabled?:
         duration: 0.01,
       })
     }
+    return () => {
+      // Close the toast when the component unmounts, even if the condition is still true
+      if (wasEnabled && params.enabled) {
+        toast.push({
+          ...params,
+          // Note: @sanity/ui fallbacks to the default duration of 4s in case of falsey values
+          duration: 0.01,
+        })
+      }
+    }
```

Comment:
This will close the toast when the component unmounts, preventing it to be forever present